src/main/frontend/handler/history.cljs

```diff
@@ -8,13 +8,13 @@
             [goog.dom :as gdom]))
 
 (defn restore-cursor!
-  [{:keys [last-edit-block container pos]}]
+  [{:keys [last-edit-block container pos end-pos]} undo?]
```

Comment:
Both the start `pos` and `end-pos` are stored; the start pos will be stored when the user starts to write anything in the editor, and the `end-pos` will be the final position when transacting to the DB.

This makes undo/redo much user-friendly, undo will go to the start pos and redo will go to the end-pos.